### PR TITLE
test : Added test for --payload option to upload archives

### PR DESCRIPTION
**Pull request by Archana-PandeyM**
1- test to verify that a pre-collected insights-archive can be uploaded 
2- test to verify wrong content-type
3- test to verify upload of a very large file

### CODE REVIEW

File: integration-tests/test_upload.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    insights_client.register()
+    assert conftest.loop_until(lambda: insights_client.is_registered)
+
+    # Running insights-client in offline mode to generate archive and save at tmp dir
+    insights_client.run(f"--output-file={archive_location}")
+
+    # Running insights-client --payload with --content-type to upload archive
+    # collected in previous step
+    upload_result = insights_client.run(
+        f"--payload={archive_location}", "--content-type=gz"
```
Comment (m-horky) [inline integration-tests/test_upload.py:25]: Maybe we should use the proper Advisor content type, as in
https://github.com/RedHatInsights/insights-core/blob/7069995dfc83783666bd967559b3f5810ffb8149/insights/client/connection.py#L372

```py
"--content-type=application/vnd.redhat.advisor.collection+tgz"
```

if we want to ensure the archive will be treated as such.

**Review by m-horky — COMMENTED** [review]
No major notes, looks good.

File: integration-tests/test_upload.py (hunk 1 above)
Comment (Archana-PandeyM) [reply integration-tests/test_upload.py:25]: I understand your point here, but the test are automated in a way customer would use the tool. I found in Insights-client user guides :
--content-type=TYPE
Set the content-type for the PAYLOAD file. Type can be gz, bz2, xz, and none. The TYPE must match the --compressor used with the PAYLOAD.

Also using the types gz, xz and bz2 processes the archive successfully. Let me know if I am missing something here.

**Review by m-horky — APPROVED** [review]
LGTM

**Pull request merged**